lib/report.c

```diff
@@ -407,28 +412,32 @@ sr_report_from_json(struct sr_json_value *root, char **error_message)
     }
 
     /* Authentication entries. */
-    struct sr_json_value *extra = json_element(root, "auth");
-    if (extra)
+    json_object *extra;
+    if (json_object_object_get_ex(root, "auth", &extra))
     {
-        if (!JSON_CHECK_TYPE(extra, SR_JSON_OBJECT, "auth"))
-            goto fail;
+        lh_table *object;
 
-        const unsigned children = json_object_children_count(extra);
+        if (!json_check_type(extra, json_type_object, "auth", error_message))
+            goto fail;
 
         /* from the last children down to the first for easier testing :)
          * keep it as it is as long as sr_report_add_auth() does LIFO */
-        for (unsigned i = 1; i <= children; ++i)
+        object = json_object_get_object(extra);
+        for (struct lh_entry *entry = object->tail; NULL != entry; entry = entry->prev)
```

Comment:
There's an [object iterator interface](https://json-c.github.io/json-c/json-c-0.13.1/doc/html/json__object__iterator_8h.html) which seems like a better approach than meddling with the internals.

Reply:
Can it go in reverse?

Reply:
Hm, right, it doesn't seem to. Though it appears that order isn't critical for `auth_entries`, it's not a huge issue either.